state/accounts/esdtAsBalance.go

```diff
@@ -75,43 +78,45 @@ func (e *esdtAsBalance) SubFromBalance(accountDataHandler vmcommon.AccountDataHa
 	}
 
 	esdtData.Value.Set(newBalance)
-	err = e.saveESDTData(accountDataHandler, esdtData)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return e.saveESDTData(accountDataHandler, esdtData)
 }
 
 func (e *esdtAsBalance) getESDTData(accountDataHandler vmcommon.AccountDataHandler) (*esdt.ESDigitalToken, error) {
-	esdtData := &esdt.ESDigitalToken{Value: big.NewInt(0), Type: uint32(core.Fungible)}
 	marshaledData, _, err := accountDataHandler.RetrieveValue(e.keyPrefix)
 	if err != nil {
-		return nil, err
+		log.Trace("esdtAsBalance.getESDTData could not load account token", "error", err)
+		return createEmptyESDT(), nil
 	}
 
+	esdtData := createEmptyESDT()
```

Comment:
this is not needed because you have the extra checks on L98. You can have this:
```GO
esdtData := &esdt.ESDigitalToken{}
```

Reply:
Refactored as suggested